src/libraries/Common/src/System/Net/ContextAwareResult.cs

```diff
@@ -158,7 +159,7 @@ internal ExecutionContext ContextCopy
                     {
                         NetEventSource.Fail(this, "Must lock (StartPostingAsyncOp()) { ... FinishPostingAsyncOp(); } when calling ContextCopy (unless it's only called after FinishPostingAsyncOp).");
```

Comment:
I think we can add to this method declaration [DoesNotReturn], then we can avoid `!` below.

Reply:
I initially had that, but then backed off since this method does return, so I thought it was more correct to not decorate it with `[DoesNotReturn]`.

I can go either way here. I'll add the attribute.

Reply:
Doesn't it always call Debug.Fail? From a nullability analysis perspective then, it would seem to be just as much a DoesNotReturn as Debug.Fail,no?

Reply:
Good point. I didn't realize `Debug.Fail` had `[DoesNotReturn]`. Fixed.